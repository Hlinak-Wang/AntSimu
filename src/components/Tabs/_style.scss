.tabs-top, .tabs-bottom {
  @include tabs-basic-style(0 3rem, none, row);

  &.tabs-inline {
    @include tabs-inline-style(inline-block, 0.5rem 0, 0 2rem 0 0);
  }
  
  &.tabs-card {
    @include basic-tabs-card-style(inline-block, 0.5rem 0.8rem, 0 0.1rem 0 0);
  }

  .prev-button-container {
    position: absolute;
    width: 32px;
    height: 100%;
    left: 0;
  }
  
  .next-button-container {
    position: absolute;
    width: 32px;
    height: 100%;
    right: 0;
  }

  .prev-button {
    left: 50%;
    top: 50%;
    &::before {
      transform: rotate(-45deg) translateX(3px);
      transition: transform 0.3s ease-in-out, background 0.3s ease-in-out;
    }
  
    &::after {
      transform: rotate(45deg) translateX(3px);
      transition: transform 0.3s ease-in-out, background 0.3s ease-in-out;
    }
  }

  .next-button {
    right: 50%;
    top: 50%;
    &::before {
      transform: rotate(-45deg) translateX(-3px);
      transition: transform 0.3s ease-in-out, background 0.3s ease-in-out;
    } 
  
    &::after {
      transform: rotate(45deg) translateX(-3px);
      transition: transform 0.3s ease-in-out, background 0.3s ease-in-out;
    }
  }
}

.tabs-top {
  @include tabs-label-style(none, bottom, bottom);
  &.tabs-card {
    @include tabs-card-direction-style(bottom, 2px 2px 0 0, 0.5rem 1.2rem);
  }
  
  .tabs-slider {
    bottom: -1px;
    left:0;
    height: 2px;
  }
}

.tabs-bottom {
  @include tabs-label-style(none, top, top);
  &.tabs-card {
    @include tabs-card-direction-style(top, 0 0 2px 2px, 0.5rem 1.2rem);
  }

  .tabs-slider {
    top: -1px;
    left:0;
    height: 2px;
  }
}

.tabs-left, .tabs-right {
  @include tabs-basic-style(2rem 0, left, column);

  &.tabs-inline {
    @include tabs-inline-style(block, 0.25rem 1rem, 0 0 1rem);
  }
  
  &.tabs-card {
    @include basic-tabs-card-style(block, 0.25rem 1rem, 0 0 0.1rem);
  }

  .prev-button-container {
    position: absolute;
    width: 100%;
    height: 32px;
    top: 0;
  }
  
  .next-button-container {
    position: absolute;
    width: 100%;
    height: 32px;
    bottom: 0;
  }

  .prev-button {
    top: 50%;
    left: 50%;
    margin-left: -3.5px;
    &::before {
      transform: rotate(-45deg) translateX(-3px);
    }
  
    &::after {
      transform: rotate(45deg) translateX(3px);
    }
  }

  .next-button {
    left: 50%;
    top: 50%;
    margin-left: -3.5px;
    &::before {
      transform: rotate(-45deg) translateX(3px);
    }
  
    &::after {
      transform: rotate(45deg) translateX(-3px);
    }
  }
}

.tabs-left {
  @include tabs-label-style(left, right, right);
  &.tabs-card {
    @include tabs-card-direction-style(right, 2px 0 0 2px, 0.6rem 0.3rem);
  }
 
  .tabs-slider {
    top: 0;
    right:-1px;
    width: 2px;
  }
}

.tabs-right {
  @include tabs-label-style(right, left, left);
  &.tabs-card {
    @include tabs-card-direction-style(left, 0 2px 2px 0, 0.6rem 0.3rem);
  }

  .tabs-slider {
    top:0;
    left:-1px;
    width: 2px;
  }
}

.tabs-label {
  display: inline-block;
  flex-wrap: nowrap;
  white-space: nowrap;
  list-style: none;
  padding: 0;
  box-sizing: border-box;
  margin: 0;
  position: relative;
  transition: transform 0.5s ease-in-out;
  .tabs-slider {
    transition: transform 0.3s ease-in-out, width 0.4s ease-out, color 0.6s ease-in-out
  }
  .label-item {
    box-sizing: border-box;
    white-space: nowrap;
    cursor: pointer;
    position: relative;
    transition: color 0.3s ease-in-out;
    &:hover {
      color: $primary;
    }
  }
}

.tabs-small {
  .tabs-label {
    font-size: 14px;
  }
}

.tabs-default {
  .tabs-label {
    font-size: 16px;
  }
}

.tabs-large {
  .tabs-label {
    font-size: 18px;
  }
}

.tabs-content {
  display: none;
  &.active {
    display: block;
  }
}

.tabs-close-btn {
  color: $gray-7;
  width: 16px;
  height: 100%;
  float: right;
  font-size: 16px;
  cursor: pointer;
  margin-left: 1rem;
  margin-right: -0.8rem;
  vertical-align: middle;
  overflow: hidden;
  &:hover {
    color: $black;
  }
}