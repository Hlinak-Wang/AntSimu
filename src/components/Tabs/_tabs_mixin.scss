@mixin tabs-basic-style(
  $container-padding,
  $content-float,
  $flex-direction
) {
  display: flex;
  .tabs-labels-container {
    display: flex;
    flex-wrap: wrap;
    flex-direction: $flex-direction;
    align-items: center;
    height: 100%;
  }

  .scroll-container {
    position: relative;
    height: 100%;
    display: block;
    overflow: hidden;
    padding: $container-padding;
    flex:1;
  }

  .tabs-content-container {
    height: 100%;
    width: 100%;
    display: block;
    position: relative;
    float: $content-float;
    overflow: auto;
  }

  .show-container {
    overflow: hidden;
    height: 100%;
    position: relative;
    &::before {
      position: absolute;
      top: 0;
      right: 0;
      left: 0;
      bottom: 0;
      content: '';
      box-shadow: inset 15px 0 15px -15px #fff,
                  inset -15px 0 15px -15px #fff;
      pointer-events: none;
      z-index: 2;
    }
  }

  .prev-button-container, .next-button-container {
    cursor: pointer;
    &.tabs-btn-diabled {
      cursor: not-allowed;
      .prev-button, .next-button {
        
        &::before {
          background: $gray-5;
          transform: rotate(-45deg);
        }
    
        &::after {
          background: $gray-5;
          transform: rotate(45deg);
        }
      }
    }
  }

  .prev-button, .next-button {
    position: absolute;
    
    &::before, &::after{
      position: absolute;
      content: "";
      display: block;
      width: 7px;
      height: 1.5px;
      background: black;
      transition: transform 0.3s ease-in-out, background 0.3s ease-in-out;
    }
  }
}

@mixin tabs-label-style(
  $label-float,
  $label-margin-direction,
  $label-border-direction
) {
  .tabs-labels-container {
    float: $label-float;
    margin-#{$label-margin-direction}: 1rem;
  }

  .tabs-label {
    border-#{$label-border-direction}: 1px solid $gray-3;
  }
}

@mixin tabs-inline-style(
  $label-item-display,
  $label-item-padding,
  $label-item-maring
) {
  .label-item {
    display: $label-item-display;
    padding: $label-item-padding;
    margin: $label-item-maring;
    transition: color 0.3s ease-in, padding 0.3s ease-in-out;
    &.active {
      color: $primary;
    }

    &.disabled {
      color: $gray-5;
      pointer-events: none;
    }
  }

  .tabs-slider {
    display: block;
    position: absolute;
    z-index: 1;
    background: $primary;
    transition: transform 0.3s ease-in-out, width 0.4s ease-out, color 0.6s ease-in-out
  }
}

@mixin basic-tabs-card-style(
  $label-item-display,
  $label-item-padding,
  $label-item-maring
) {
  .label-item {
    display: $label-item-display;
    padding: $label-item-padding;
    margin: $label-item-maring;
    background: $gray-1;
    border: 1px solid $gray-3;
    transition: color 0.3s ease-in, padding 0.3s ease-in-out;
    &.active {
      color: $primary; 
      background: $white;
    }
  }

  .tabs-slider {
    display: block;
    position: absolute;
    z-index: 1;
    background: transparent;
    transition: transform 0.3s ease-in-out, width 0.4s ease-out, color 0.6s ease-in-out
  } 
}

@mixin tabs-card-direction-style(
  $effect-direction,
  $border-radius,
  $padding
) {
  .show-container {
    margin-#{$effect-direction}: -1px;
    .tabs-label {
      border-#{$effect-direction}: 1px solid $gray-3;
      .label-item{
        border-radius: $border-radius;
        margin-#{$effect-direction}: -1px;
        &.active {
          border-#{$effect-direction}: 1px solid $white;
          padding: $padding;
        }
      }
    }
  }
}