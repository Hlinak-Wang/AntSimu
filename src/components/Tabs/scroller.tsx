import React, { FC, useCallback, useState, useRef, useEffect } from 'react';
import SliderBar from './sliderBar';
import classnames from 'classnames';

interface IScroller {
  direction: 'vertical' | 'horizontal';
  activeRef: any;
  prevButton?: React.ReactNode;
  nextButton?: React.ReactNode;
}

const Scroller: React.FC<IScroller> = ({ direction, activeRef, children }) => {

  const containRef = useRef<HTMLDivElement | null>(null);
  const scrollRef = useRef<HTMLDivElement | null>(null);
  const [shiftX, setShiftX] = useState<number>(0);
  const [shiftY, setShiftY] = useState<number>(0);

  const scrollEventCallback = useCallback((e: WheelEvent) => {
    handleScroll(e)
  }, [shiftX, shiftY]);

  function handleScroll(e: WheelEvent) {

    if (e.deltaY <= 0) {
      // scroll top
      if (prevAbleStatus()) {
        e.preventDefault();
      }
      prev();
    } else if (e.deltaY > 0){
      // scroll down
      if (nextAbleStatus()) {
        e.preventDefault();
      }
      next();
    }
  }

  useEffect(() => {
    if (scrollRef.current) {
      scrollRef.current.addEventListener('wheel', scrollEventCallback, { passive: false });
    }
    
    return () => {
      if (scrollRef.current) {
        scrollRef.current.removeEventListener('wheel', scrollEventCallback);
      }
    }
  }, [shiftX, shiftY])


  useEffect(() => {

    if (activeRef && scrollRef.current && containRef.current) {
      if (direction === 'horizontal') {
        if (activeRef.clientWidth + activeRef.offsetLeft + shiftX > containRef.current.clientWidth) {
          setShiftX(-activeRef.offsetLeft);
        } else if (activeRef.offsetLeft < -shiftX) {
          setShiftX(-activeRef.offsetLeft);
        }
      } else if (direction === 'vertical') {

        if (activeRef.clientHeight + activeRef.offsetTop + shiftY > containRef.current.clientHeight) {
          setShiftY(-activeRef.offsetTop);
        } else if (activeRef.offsetTop < -shiftY) {
          setShiftY(-activeRef.offsetTop); 
        }
      }
    }
  }, [activeRef]);

  function prevAbleStatus() {
    if (shiftX || shiftY) {
      return true;
    } else {
      return false;
    }
  }

  function nextAbleStatus() {
    if (containRef.current && scrollRef.current) {
      if (direction === 'horizontal') {
        // check horizontal
        if (shiftX - containRef.current.clientWidth <= -scrollRef.current.clientWidth) {
          return false;
        }
      } else {
        // check vertical
        if (shiftY - containRef.current.clientHeight <= -scrollRef.current.clientHeight) {
          console.log("ver")
          return false;
        }
      }
    } 
    return true;
  }
 
  function prev() {
    if (!containRef.current) {
      console.error("can't get container DOM");
      return;
    }

    if (direction === 'horizontal' && containRef.current) {
      if (shiftX + containRef.current.clientWidth > 0) {
        setShiftX(0);
      } else if (shiftX) {
        setShiftX(shiftX + containRef.current.clientWidth);
      }
    } else if (direction === 'vertical') {
      if (shiftY + containRef.current.clientHeight > 0) {
        setShiftY(0);
      } else if (shiftY) {
        setShiftY(shiftY + containRef.current.clientHeight);
      }
    }
  }

  function next() {
    if (!containRef.current) {
      console.error("can't get container DOM");
      return;
    }

    if (!scrollRef.current) {
      console.error("can't get scroller DOM");
      return;
    }

    if (direction === 'horizontal') {
      if (shiftX - containRef.current.clientWidth <= -scrollRef.current.clientWidth) {
        return;
      }

      if (shiftX - 2 * containRef.current.clientWidth < -scrollRef.current.clientWidth) {
        setShiftX(containRef.current.clientWidth - scrollRef.current.clientWidth);
      } else if (shiftX - containRef.current.clientWidth > -scrollRef.current.clientWidth) {
        setShiftX(shiftX - containRef.current.clientWidth);
      }
    } else if (direction === 'vertical') {
      if ( shiftY - containRef.current.clientHeight <= -scrollRef.current.clientHeight) {
        return;
      }

      if (shiftY - 2 * containRef.current.clientHeight < -scrollRef.current.clientHeight) {
        setShiftY(containRef.current.clientHeight - scrollRef.current.clientHeight);
      } else if (shiftY - containRef.current.clientHeight > -scrollRef.current.clientHeight) {
        setShiftY(shiftY - containRef.current.clientHeight);
      } 
    }
  }

  const prevCls = classnames("prev-button-container", {
    "tabs-btn-diabled": !prevAbleStatus()
  })

  const nextCls = classnames("next-button-container", {
    "tabs-btn-diabled": !nextAbleStatus()
  })

  return (
    <div className="scroll-container">
      <div onClick={prev} className={prevCls}>
        <div className="prev-button"/>
      </div>
      <div onClick={next} className={nextCls}>
        <div className="next-button"/>
      </div>
      <div className="show-container" ref={containRef}>
        <div className="tabs-label" ref={scrollRef} style={{transform: `translate3d(${shiftX}px,${shiftY}px,0px)`}}>
          {children}
          <SliderBar activeRef={activeRef} direction={direction}/>
        </div>
      </div>
    </div>
  );
};

export default Scroller;