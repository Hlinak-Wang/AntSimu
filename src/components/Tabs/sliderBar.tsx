import React, { useState, useEffect } from 'react';

interface ISliderBar {
  activeRef: React.MutableRefObject<HTMLDivElement | null>;
  direction: 'horizontal' | 'vertical'
} 

const SliderBar: React.FC<ISliderBar> = ({ activeRef, direction }) => {

  const [sliderCSS, setSliderCSS] = useState<React.CSSProperties>({});

  useEffect(() => {
    if (activeRef.current) {
      if (direction === 'horizontal') {
        setSliderCSS({
          width: `${activeRef.current.clientWidth}px`,
          transform: `translate3d(${activeRef.current.offsetLeft}px,0,0)`,
        })
      } else if (direction === 'vertical') {
        setSliderCSS({
          height: `${activeRef.current.clientHeight}px`,
          transform: `translate3d(0,${activeRef.current.offsetTop}px,0)`,
        })
      }
    }
  }, [activeRef.current])

  return (
    <div className="tabs-slider" style={sliderCSS} />
  );
};

export default SliderBar;