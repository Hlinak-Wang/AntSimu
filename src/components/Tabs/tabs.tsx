import React, { FC, createContext, useState, useEffect } from 'react';
import classnames from 'classnames';
import Scroller from './scroller';
import TabsContent from './tabsContent'
import TabsLabel from './tabsLabel';

type TabsType = 'inline' | 'card';
type TabsSize = 'small' | 'default' | 'large';
type TabPosition = 'top' | 'bottom' | 'right' | 'left';

export interface TabsItemProps {
  label: React.ReactNode;
  closable?: boolean;
  style?:React.CSSProperties;
  disabled?:boolean;
  key: string;
  children:React.ReactNode;
}

interface TabsProps {
  disabled?:boolean;
  children: React.ReactNode;
  type?: TabsType;
  size?: TabsSize;
  defaultActiveKey?: string;
  tabBarExtraContent?: React.ReactNode;
  TabPosition?: TabPosition;
  onTabClick?: Function;
  onNextClick?: Function;
  onPrevClick?: Function;
  style?: React.CSSProperties;
  closable?: boolean;
  onEdit?: (targetIndex: string, action: 'add' | 'remove') => void;
}

export interface ITabsContext {
  activeKey?: string;
  TabPosition?: TabPosition;
  closable?: boolean;
  tabClickRes?: (index: string) => void;
}

export class TabsItem extends React.Component<TabsItemProps>{}

export const TabsContext = createContext<ITabsContext>({});

const Tabs: FC<TabsProps> = (props) => {
  
  const { 
    type, 
    size, 
    defaultActiveKey, 
    tabBarExtraContent, 
    TabPosition, 
    closable, 
    onEdit, 
    onTabClick, 
    style, 
    children 
  } = props;

  const [activeKey, setSelect] = useState(defaultActiveKey);

  function handleSelect(index: string) {
    setSelect(index);
    onTabClick && onTabClick();
  }

  const context = {
    activeKey,
    TabPosition,
    closable,
    tabClickRes: handleSelect
  }

  const removeTabs = (targetKey: string, e: React.MouseEvent<HTMLElement>) => {
    e.stopPropagation();
    if (onEdit) {
      onEdit(targetKey, 'remove')
    }
  }

  const addTabs = (targetKey: string) => {
    if (onEdit) {
      onEdit(targetKey, 'add')
    }
  }

  useEffect(() => {
    React.Children.forEach(children, (child, index) => {
      const childElement = child as React.FunctionComponentElement<TabsItemProps>;
      const displayName = childElement.type.displayName || childElement.type.name;
      if (displayName !== 'TabsItem') {
        console.error("Tabs has a child which is not TabsItem component")
      }
    })
  }, [children])

  const itemClass = classnames('tabs', 
    `tabs-${type}`, 
    `tabs-${size}`, 
    `tabs-${TabPosition}`
  )

  return (
    <div className={itemClass} style={style}>
      <TabsContext.Provider value={context}>
        {
          TabPosition === 'bottom' || TabPosition === 'right'
          ? <>
            <TabsContent 
              items={children}
            />
            <div className="tabs-labels-container">
              {tabBarExtraContent}
              <TabsLabel 
                items={children}
              />
            </div>
            
          </>
          : <>
            <div className="tabs-labels-container">
              {tabBarExtraContent}
              <TabsLabel 
                items={children}
              />
            </div>
            <TabsContent 
              items={children}
            />
          </>
        }
      </TabsContext.Provider>
    </div>
  )
}

Tabs.defaultProps = {
  type: 'inline',
  size: 'default',
  TabPosition: 'top',
  closable: false,
}

export default Tabs;