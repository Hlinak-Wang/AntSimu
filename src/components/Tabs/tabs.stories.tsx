import React from 'react';
import Tabs, {TabsItem} from './tabs';
import { storiesOf } from '@storybook/react';
import { actions } from '@storybook/addon-actions';
import Button from '../Button/button';

const arr = (props: Object) => (
  <Tabs {...props}>
  <TabsItem label="1" key="1" disabled>
    label 1
  </TabsItem>
  <TabsItem label="2" key="2" disabled>
    label 2
  </TabsItem>
  <TabsItem label="3" key="3">
    label 3
  </TabsItem>
  <TabsItem label="4" key="4">
    label 4
  </TabsItem>
  <TabsItem label="5" key="5">
    label 5
  </TabsItem>
  <TabsItem label="6" key="6">
    label 6
  </TabsItem>
  <TabsItem label="7" key="7">
    label 7
  </TabsItem>
  <TabsItem label="6" key="8">
    label 6
  </TabsItem>
  <TabsItem label="7" key="9">
    label 7
  </TabsItem>
  <TabsItem label="6" key="10">
    label 6
  </TabsItem>
  <TabsItem label="7" key="11">
    label 7
  </TabsItem>
  <TabsItem label="4" key="12">
    label 4
  </TabsItem>
  <TabsItem label="5" key="13">
    label 5
  </TabsItem>
  <TabsItem label="6" key="14">
    label 6
  </TabsItem>
  <TabsItem label="7" key="15">
    label 7
  </TabsItem>
  <TabsItem label="6" key="16">
    label 6
  </TabsItem>
  <TabsItem label="7" key="17">
    label 7
  </TabsItem>
  <TabsItem label="4" key="18">
    label 4
  </TabsItem>
  <TabsItem label="5" key="19">
    label 5
  </TabsItem>
  <TabsItem label="6" key="20">
    label 6
  </TabsItem>
  <TabsItem label="7" key="21">
    label 7
  </TabsItem>
  <TabsItem label="6" key="22">
    label 6
  </TabsItem>
  <TabsItem label="7" key="23">
    label 7
  </TabsItem>
  <TabsItem label="6" key="24">
    label 6
  </TabsItem>
  <TabsItem label="7" key="25">
    label 7
  </TabsItem>
  <TabsItem label="4" key="26">
    label 4
  </TabsItem>
  <TabsItem label="5" key="27">
    label 5
  </TabsItem>
  <TabsItem label="6" key="28">
    label 6
  </TabsItem>
  <TabsItem label="7" key="29">
    label 7
  </TabsItem>
  <TabsItem label="6" key="30">
    label 6
  </TabsItem>
  <TabsItem label="7" key="31">
    label 7
  </TabsItem>
  <TabsItem label="4" key="32">
    label 4
  </TabsItem>
  <TabsItem label="5" key="33">
    label 5
  </TabsItem>
  <TabsItem label="6" key="34">
    label 6
  </TabsItem>
  <TabsItem label="7" key="35">
    label 7
  </TabsItem>
  </Tabs>
);
  
const button = <><Button>tesa</Button></>;

storiesOf('Tabs', module)
  .add('基础样式', () => (
    arr({TabPosition: "top", defaultActiveKey: "10", tabBarExtraContent:button})
  ))
  .add('可设置方向', () => (
    arr({TabPosition: "bottom", tabBarExtraContent:button})
  ))
  .add('left', () => (
    arr({TabPosition:"left", style: {height: "500px"}, tabBarExtraContent:button})
  ))
  .add('right', () => (
    arr({TabPosition:"right", style: {height: "500px"}, tabBarExtraContent:button})
  ))