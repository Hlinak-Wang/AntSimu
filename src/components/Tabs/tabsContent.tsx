import React, { FC, useContext, useState, useCallback } from 'react';
import classnames from 'classnames';
import { TabsContext, TabsItemProps } from './tabs';
import { CSSTransition } from 'react-transition-group';

export interface TabsContentProps {
  items: React.ReactNode;
}

const TabsContent: FC<TabsContentProps> = ({ items }) => {

  const context = useContext(TabsContext);

  return (
    <div className="tabs-content-container">
        {
          React.Children.map(items , child => {
            const childElement = child as React.FunctionComponentElement<TabsItemProps>;
            const displayName = childElement.type.displayName || childElement.type.name;
            if (displayName !== 'TabsItem') {
              console.error("Tabs has a child which is not TabsItem component")
            }

            const cls = classnames("tabs-content", {
              "tabs-content-active": context.activeKey === childElement.key
            })

            return (
                <div 
                  className={cls} style={childElement.props.style ? childElement.props.style : {}}
                >
                {childElement.props.children}
                </div>
            );
          })
        } 
    </div>
  )
}


export default TabsContent;