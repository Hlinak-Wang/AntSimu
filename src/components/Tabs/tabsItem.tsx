import React, { useContext } from 'react';
import classnames from 'classnames';
import { TabsContext } from './tabs';

export interface TabsItemProps {
  label: React.ReactNode;
  index?: number;
  children: React.ReactNode;
}

const TabsItem: React.FC<TabsItemProps> = ({index, children}) => {

  const context = useContext(TabsContext);

  const itemClass = classnames('tabs-content', {
    'active': context.index === index
  })

  return (
    <div className={itemClass}>
      {children}
    </div>
  )
}

export default TabsItem;