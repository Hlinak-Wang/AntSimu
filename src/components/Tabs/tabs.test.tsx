import React from 'react';
import Tabs from './tabs';
import TabsItem from './tabsItem';
import { render, RenderResult, fireEvent, cleanup } from '@testing-library/react';

describe('basic test', () => {
  it('should render correctly for different props TabPosition', () => {
    const wrapp = render(<Tabs type="inline" TabPosition="top">
      <TabsItem label="1">hello</TabsItem>
      <TabsItem label="2">123</TabsItem>
    </Tabs>)
    expect(wrapp).toMatchSnapshot();
  })
})