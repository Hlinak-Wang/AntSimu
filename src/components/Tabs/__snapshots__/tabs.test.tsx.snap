// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test should render correctly for different props TabPosition 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tabs tabs-inline tabs-default tabs-top"
      >
        <div
          class="tabs-label-container"
        >
          <div
            class="prev-button tabs-btn-diabled"
          >
            prev
          </div>
          <div
            class="next-button tabs-btn-diabled"
          >
            next
          </div>
          <div
            class="show-container"
          >
            <div
              class="tabs-label"
              style="transform: translate3d(0px,0px,0px);"
            >
              <div
                class="label-item active"
              >
                1
              </div>
              <div
                class="label-item"
              >
                2
              </div>
              <div
                class="tabs-slider"
                style="width: 0px; transform: translate3d(0px,0,0);"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tabs tabs-inline tabs-default tabs-top"
    >
      <div
        class="tabs-label-container"
      >
        <div
          class="prev-button tabs-btn-diabled"
        >
          prev
        </div>
        <div
          class="next-button tabs-btn-diabled"
        >
          next
        </div>
        <div
          class="show-container"
        >
          <div
            class="tabs-label"
            style="transform: translate3d(0px,0px,0px);"
          >
            <div
              class="label-item active"
            >
              1
            </div>
            <div
              class="label-item"
            >
              2
            </div>
            <div
              class="tabs-slider"
              style="width: 0px; transform: translate3d(0px,0,0);"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
